Sources/AIProxy/OpenAI/OpenAIChatCompletionRequestBody.swift

```diff
@@ -186,7 +186,7 @@ public enum OpenAIChatCompletionContentPart: Encodable {
 
     /// The URL is a "local URL" containing base64 encoded image data. See the helper `AIProxy.openaiEncodedImage`
     /// to construct this URL.
-    case imageURL(URL)
+    case imageURL(url: URL, detail: String?)
```

Comment:
This will break existing callers. Notice that this is `public enum`. We should assume that anything in the lib marked as `public` has existing callers, and therefore we should try not to break the contract unless necessary. We are still pre-version 1, so we can break contracts, but it is always annoying to the developer to pull in a new version and have to update the call sites so we do it sparingly.

You can add the feature you need by giving this case the form:
```
case imageURL(URL, detail: String? = nil)
```

This lets people opt-in to the detail string, while remaining backwards compatible.

Couple other things you should address:

1. Add a docstring to describe what the `detail` param does
2. Since `detail` can only be one of three strings, I would model it as an enum:

```
extension OpenAIChatCompletionContentPart {
  public enum Detail: String, Encodable {
    case auto
    case low
    case high
  }
}
```

Then this signature becomes:

```
case imageURL(URL, detail: Detail)
```

And callers use `.imageURL(someURL, detail: .low)`, for example.

Also add a docstring to each case that is taken from here: https://platform.openai.com/docs/guides/vision/low-or-high-fidelity-image-understanding.

I think I was a bit too lax in the OpenAI encodables, but we should really have docstrings on every controllable param. Our customers tend to not read the provider's sites, and instead rely on the lib to see what's possible.